glide/blog.md

```diff
@@ -0,0 +1,147 @@
+TLDR; [Glide](glide-browser.app) is a Firefox-based, [modal](#modes), and [malleable](https://glide-browser.app/config)[^1] web browser for the motivated tinkerer.
+
+##### invisible-heading
+
+[Download](https://glide-browser.app/#download) - [Docs](https://glide-browser.app/) - [Cookbook](https://glide-browser.app/cookbook) - [Source](https://github.com/glide-browser/glide)
+
+---
+
+<br>
+
+I believe browsers should be hackable, just like your editor.
+
+```typescript
+glide.keymaps.set("normal", "gC", async () => {
+  // extract the owner and repo from a url like 'https://github.com/glide-browser/glide'
+  const [owner, repo] = glide.ctx.url.pathname.split("/").slice(1, 3);
+  if (!owner || !repo) throw new Error("current URL is not a github repo");
+
+  // * clone the current github repo to ~/github.com/$owner/$repo
+  // * start kitty with neovim open at the cloned repo
+  const repo_path = glide.path.join(glide.path.home_dir, "github.com", owner, repo);
+  await glide.process.execute("gh", [ "repo", "clone", glide.ctx.url, repo_path]);
+  await glide.process.execute("kitty", ["-d", repo_path, "nvim"], { cwd: repo_path });
+}, { description: "open the GitHub repo in the focused tab in Neovim" });
+```
+
+> Example Glide keymapping to clone the GitHub repo in the current tab and open it in [Kitty](https://sw.kovidgoyal.net/kitty/) + [Neovim](https://neovim.io/).
+
+This snippet captures exactly the kind of functionality I personally want from a browser; extreme customisability so that I can make my daily workflows a joy.
+
+### What's wrong with the status quo?
+
+Most browsers only allow you to customise shortcuts for certain predefined actions, and do not allow you to define your own entirely custom keymappings without going through a web extension; and even then support for writing your own code is very limited.
+
+Existing web extensions[^2] do offer similar functionality to Glide but are artificially restricted in scope by security constraints[^3] imposed on all web extensions.
+
+For example, extensions are completely disabled on certain URIs, e.g. [addons.mozilla.org](https://addons.mozilla.org), the json file viewer, etc. Which means that relying on extension defined mappings on those pages will not work.
+
+There are also *many* useful APIs that are not provided to web extensions, such as arbitrary file writes, spawning processes, and setting prefs.
```

Comment:
I think an interesting angle here is why is it that web extensions are limited.

My take is that web extension limitations are there mostly to protect the user from the extension writers. In Glide, you are trusted to write the code for yourself.

Reply:
oh you do write this below, me dumb